src/content/contribute/writing-a-loader.md

```diff
@@ -370,9 +370,9 @@ import compiler from './compiler.js';
 
 test('Inserts name and outputs JavaScript', async () => {
   const stats = await compiler('example.txt', { name: 'Alice' });
-  const output = stats.toJson().modules[0].source;
+  const output = stats.toJson({source: true}).modules[0].source;
 
-  expect(output).toBe('export default "Hey Alice!\"');
+  expect(output).toBe('export default "Hey Alice!\\n"');
```

Comment:
Can you double check this one? I've tested locally and it's not needed:

![image](https://user-images.githubusercontent.com/1091472/97799616-2bb00880-1c6a-11eb-82c7-f43160d0a1ae.png)

Just in case you want to have a look, here's the [repository](https://github.com/chenxsan/webpack-write-a-loader) I used to test.

Reply:
It's hard to really tell.  Line endings are not visible in single-line code listings.  The listing in your screenshot might have a line ending; or it might not.  You can't tell with a single line file.  The vi editor, by default, will automatically create a line ending on the end, even though I don't press enter.  I can create a file with no line endings with the `echo -n` option:

```
echo -n 'Hey [name]!' > example.txt
```

Perhaps this, too, could be added to the instructions.  Then we could remove the `\n` on line 375 above.

Reply:
How about adding an explicit comment to that example, for instance:

```
Hey [name]! // no line ending
```
And update the assertion to:

```
expect(output).toBe('export default "Hey Alice! // no line ending\"');
```

Reply:
Then you're factually correct.  I just worry about people with editors that automatically add a new-line without them knowing.  But, hey, if you're far enough along to be writing webpack loaders, you can figure out whether your editor adds a gratuitous new-line before end of file.  So I'm good with it.

Reply:
I can see your worry here, as we're trying to cover two possible cases here with one assertion.